src/Walkthrough.ts

```diff
@@ -41,6 +41,12 @@ export class Walkthrough {
     nextButton.addEventListener("click", () => this.goToNextStep());
     prevButton.addEventListener("click", () => this.goToPreviousStep());
     closeButton.addEventListener("click", () => {
+      const { onExit } = this._config[this._currentStepIndex];
```

Comment:
@Raghu3130 this way you will have to give an exit function to every step in the walkthrough. Best is if you pass it in the 2nd argument to the ``start`` function, and then call it within the exit function. So that even when the user finishes on the last step it gets patched.

Reply:
You can save it under ``this._onExit`` on the class level (Walkthrough)